File: src/main/java/fr/epicanard/mapsaver/models/language/Language.java
package fr.epicanard.mapsaver.models.language;

import fr.epicanard.duckconfig.annotations.Header;

import java.util.Map;

@Header({
        "==============",
        "Language file for plugin MapSaver",
        "=============="
})
public class Language {
    public Help Help;
    public ErrorMessages ErrorMessages;
    public InfoMessages InfoMessages;
    public MapInfo MapInfo;
    public List List;
    public Map<String, String> Visibility;
}

File: src/main/scala/fr/epicanard/mapsaver/resources/language/Language.scala
package fr.epicanard.mapsaver.resources.language

import fr.epicanard.mapsaver.circe.CapitalizeConfiguration
import fr.epicanard.mapsaver.models.map.Visibility
import io.circe.{KeyDecoder, KeyEncoder}
import io.circe.generic.extras.ConfiguredJsonCodec

@ConfiguredJsonCodec
case class Language(
    help: Help,
    errorMessages: ErrorMessages,
    infoMessages: InfoMessages,
    mapInfo: MapInfo,
    list: List,
    pagination: Pagination,
    visibility: Map[Visibility, String]
)

object Language extends CapitalizeConfiguration {
  implicit val visibilityKeyDecoder: KeyDecoder[Visibility] = KeyDecoder.instance(key => Visibility.withNameOption(key))

  implicit val visibilityKeyEncoder: KeyEncoder[Visibility] = KeyEncoder.instance(visibility => visibility.entryName)
}
